Sources/BSON/BSONDecimal128.swift

```diff
@@ -0,0 +1,283 @@
+import Foundation
+import NIO
+
+public struct UInt128: Equatable, Hashable {
```

Comment:
I think this type should probably be internal. the Decimal128 type can just expose the standardized string representation.